spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/RabbitAnnotationDrivenConfiguration.java

```diff
@@ -55,8 +58,8 @@
 
 	@Bean
 	@ConditionalOnMissingBean
-	public SimpleRabbitListenerContainerFactoryConfigurer rabbitListenerContainerFactoryConfigurer() {
-		SimpleRabbitListenerContainerFactoryConfigurer configurer = new SimpleRabbitListenerContainerFactoryConfigurer();
+	public RabbitListenerContainerFactoryConfigurer rabbitListenerContainerFactoryConfigurer() {
```

Comment:
I think I'd prefer two configurer implementations with dedicated types and a base class. I understand that's quite a lot of code but that probably translates well with what that thing is configuring.